def/configmap.bzl

```diff
@@ -0,0 +1,96 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_k8s//k8s:object.bzl", "k8s_object")
+
+def _impl(ctx):
+  args = [
+    "--output=%s" % ctx.outputs.output.path,
+    "--name=%s" % ctx.label.name[:-len(".generated-yaml")],
+    "--namespace=%s" % ctx.attr.namespace,
+  ]
+  for key, value in ctx.attr.labels.items():
+    args.append("--label=%s=%s" % (key, value))
+
+  # Build the {string: label} dict
+  targets = {}
+  for i, t in enumerate(ctx.attr.data_strings):
+    targets[t] = ctx.attr.data_labels[i]
+
+  for name, label in ctx.attr.data.items():
```

Comment:
Here `label` really means `target`?

Reply:
Label is just a "//foo:bar" string here, which gets converted into a target by targets["//foo:bar"]